hydroflow_datalog_core/src/join_plan.rs

```diff
@@ -409,12 +422,24 @@ pub fn expand_join_plan(
                 parse_quote!(|kv: (#key_type, (#left_type, #right_type))| (#(#flattened_tuple_elems, )*))
             };
 
+            let (lt_left, lt_right, is_persist): (syn::Lifetime, syn::Lifetime, bool) =
+                if left_expanded.persisted && !right_expanded.persisted && !is_anti {
+                    (parse_quote!('static), parse_quote!('tick), false)
+                } else if !left_expanded.persisted && right_expanded.persisted && !is_anti {
+                    (parse_quote!('tick), parse_quote!('static), false)
+                } else if left_expanded.persisted && right_expanded.persisted && !is_anti {
+                    (parse_quote!('static), parse_quote!('static), true)
+                } else {
+                    (parse_quote!('tick), parse_quote!('tick), false)
+                };
+
```

Comment:
I suppose eventually we'll want the hydroflow compiler to optimize these sorts of things a lot more which means less work here maybe.

Reply:
might be cleaner as a match
```rust
match (left_expanded.persisted, right_expanded.persisted, is_anti) {
    (true, false, false) => (parse_quote!('static), parse_quote!('tick), false),
    (false, true, false) => (parse_quote!('tick), parse_quote!('static), false),
    (true, true, false) => (parse_quote!('static), parse_quote!('static), true),
    _ => (parse_quote!('tick), parse_quote!('tick), false)
}
```
assume it compiles to the same thing probably